### DISCUSSION TITLE
Keep X number of best recordings per species  + purge Disk Behaviour

### DISCUSSION THREAD

**Discussion by DemBones78** [mcguirepr89/BirdNET-Pi#972] [Ideas]
Feature request/behavior change.

Add setting to Disk Behaviour,
Keep  X number of best recordings of each detected species.

This would effectively mark the 1 or 2 best recordings per bird saved.

I was looking at the amazing and much improved (species stats) page. 
but some of my rarer or off-session stats don't have any associated recordings because they have not been seen in months.
It would be really nice if the stat page had a recording for every species' stat.
I think my feature would be a way to achieve that.  

Thanks for your consideration! I love Bird PI

 this is my station,  it gets over 5000 recordings each day.
https://app.birdweather.com/stations/644
So the purge happens on my system at about  60
Here is an example, this drop-down lists the names of the recordings but they are all gone.
( the change should make the best recording page = to the total number of good species detected.

For those wondering, my setup is a raspberry pi 3b in my attic, with a USB extension cable running up to the gable eave of the south side of my house, with a $12 amazone lapel mic nailed to the overhang of the roof at the peak.

**Comment by DemBones78** [comment]
![bpi](https://github.com/mcguirepr89/BirdNET-Pi/assets/46800652/752e7c70-3292-422d-8f9a-f5abe3c8bf85)

**Comment by DemBones78** [comment]
![apb1](https://github.com/mcguirepr89/BirdNET-Pi/assets/46800652/39a661ec-bea1-451c-a538-03fb00ec1cd1)

**Comment by DemBones78** [comment]
just bumping this becase the ability to have one setting that Keeps 1 ( the best ) recoding for each species is really needed to make the species stats page work correctly, otherwise seasonal birds get purged

**Comment by pddpauw** [comment]
the best thing to do, would be to execute daily script (e.g. cron)
In this script, you would delete the recordings based on your defined parameter (e.g. keeping 5 recording with highest conf level per species).  You can then decide if you also wish to delete this from the DB.

**Comment by DemBones78** [comment]
ok that sounds like a good solution until that functionality is added .
what would a script like that even look like? i am not a programmer, I am barely literate .

**Comment by pddpauw** [comment]
what I did, is -on daily basis, removing the most common birds which woud typically take up to 80% of my disk space.
Only the wav files get removed; the observation stays in the database:

via crontab:
```

0 0 * * * find /home/pi/BirdSongs -type d -name '*Roodborst*' -exec rm -drfv {} +
0 0 * * * find /home/pi/BirdSongs -type d -name '*Ekster*' -exec rm -drfv {} +
0 0 * * * find /home/pi/BirdSongs -type d -name '*Gaai*' -exec rm -drfv {} +
0 0 * * * find /home/pi/BirdSongs -type d -name '*Goudhaan*' -exec rm -drfv {} +
0 0 * * * find /home/pi/BirdSongs -type d -name '*Merel*' -exec rm -drfv {} +
0 0 * * * find /home/pi/BirdSongs -type d -name '*Bosuil*' -exec rm -drfv {} +](url)
```

**Comment by DemBones78** [comment]
yeah that does not seem to do what I am looking for , I want to keep the best recording of every bird

**Reply by pddpauw** [reply]
of course not, was just as an example. As written before, you will need to write your owns script that will delete the records base on how you define "best" recording.  This script could then be executed on a daily basis, e.g. as cronjob.
You could do this as e.g. bash script or python script (I would advise the latter as it will give you more possibilities).

Just do google: python delete database record mySQL, and you will get plenty of examples, and it should be super straightforward to do this.  The tutorials describe how to open a DB, read data and remove data.

In essence you will need to select a date (e.g. yesterday in case you plan to run the script the day after), and for each species in your DB for that day, you will need to read the highest confidence level (assuming this is defining "best" for you), and then only keep this record (thus delete all others).  This will need to be repeated for all species.

Give it a go, and good luck with this!

**Comment by alexbelgium** [comment]
> Feature request/behavior change.
> 
> Add setting to Disk Behaviour, Keep X number of best recordings of each detected species.


Hi, just passing through. We've implemented that feature in the Nachtzuster fork
